pkg/cvo/sync_worker.go

```diff
@@ -435,11 +435,13 @@ func (w *SyncWorker) Update(ctx context.Context, generation int64, desired confi
 				Image:   work.Desired.Image,
 			},
 		}
+		w.work = work
 	} else {
 		oldDesired = &w.work.Desired
 	}
 
-	w.work = work
+	// save latest capability settings
+	w.work.Capabilities = work.Capabilities
 
 	if !versionEqual && oldDesired == nil {
 		klog.Infof("Propagating initial target version %v to sync worker loop in state %s.", desired, state)
```

Comment:
we may want to shift this log block to after the `w.loadUpdatedPayload` too.  We want to keep the `Ignoring detected version change...` bailout block before the `w.loadUpdatedPayload`.